src/replication/adapter/dockerhub/adapter.go

```diff
@@ -25,16 +28,16 @@ func init() {
 }
 
 func factory(registry *model.Registry) (adp.Adapter, error) {
-	client, err := NewClient(&model.Registry{
-		URL:        baseURL, // specify the URL of Docker Hub
-		Credential: registry.Credential,
-		Insecure:   registry.Insecure,
-	})
+	client, err := NewClient(registry)
 	if err != nil {
 		return nil, err
 	}
 
-	dockerRegistryAdapter, err := native.NewAdapter(registry)
+	dockerRegistryAdapter, err := native.NewAdapter(&model.Registry{
```

Comment:
what's the reason for exchanging the passed-in parameters?

Reply:
DockerHub use too different url for v2 api and others. For example, `hub.docker.io` to list namespace, delete repo, and `registry-1.docker.io` for v2 apis.